presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -248,14 +254,16 @@ public CounterStat getHighMemorySplitSource()
                 }
 
                 // verify all partition is online
-                String protectMode = partition.getParameters().get(ProtectMode.PARAMETER_NAME);
-                String partName = makePartName(table.getPartitionColumns(), partition.getValues());
-                if (protectMode != null && getProtectModeFromString(protectMode).offline) {
-                    throw new PartitionOfflineException(tableName, partName, false, null);
-                }
-                String prestoOffline = partition.getParameters().get(PRESTO_OFFLINE);
-                if (!isNullOrEmpty(prestoOffline)) {
-                    throw new PartitionOfflineException(tableName, partName, true, prestoOffline);
+                verifyOnline(
+                        tableName,
+                        Optional.of(makePartName(table.getPartitionColumns(), partition.getValues())),
+                        getProtectMode(partition),
+                        table.getParameters());
+
+                // verify all partition is readable
```

Comment:
> verify all partition is readable

what does it mean?

Reply:
@findepi : If it has flag `object_not_readable`, it will be prevented from reading from Presto side. But the comment might be a bit ambiguous. Thank you for pointing out!